src/main/java/org/eclipse/uprotocol/communication/UPayload.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.uprotocol.communication;
+
+
+import java.util.Objects;
+import java.util.Optional;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+
+import org.eclipse.uprotocol.v1.UPayloadFormat;
+
+/**
+ * Wrapper class that stores the payload as {@link UPayloadFormat}.
+ */
+public class UPayload {
+
+    private ByteString data;
```

Comment:
Only difference from ByteString and byte[] (per Google documentation) is that ByteString is immutable and byte[] is not. We don't want it to change hence ByteString.

Reply:
I am a fan of immutability, but it can be enforced by the UPayload being a Java 17 record which by default is immutable with an equals, hashcode and toString implemented (they took the idea from scala case classes)

My problem was, when I wanted to serialize the UPayload to a JSON representation, the ByteString did not play nicely with the Jackson ObjectMapper.

Another thing is that if we are already working with a google byte string, why not work with a google Message instead.
Then anyone can turn it into any kind of serialization.

Personally, I am still a fan of the simple byte[] 
It gives the best flexibility. 

Currently, I will not be able to use UPayload like this in my code and will need to wrap it with something else

Reply:
Your proposal results then in yet another copy when we build the UMessage that is passed to transport.send().